build cli issues on linux

OS: Linux Mint 19.3 based on Ubuntu 18.04 which is based on Debian.

First issue is [install-linux.txt](../blob/d7dca81b2707fe962f116b23bface18de88f4351/install-linux.txt#L9) says `make -f Makefile.freebsd` but should probably say `make -f Makefile.linux` or say why _Makefile.linux_ should not be used on linux.

Second issue is that, neither `make -f Makefile.freebsd` or `make -f Makefile.linux` work on Linux Mint 19.

After installing _build-essential libusb-dev_ and _bzip2_, `make -f Makefile.freebsd` result is:

```
rm -rf build
mkdir -p build; cp *dat build/; cp -r docs build/
g++8 -g -c libibooter.cpp -Isrc -o build/libibooter.o
make: g++8: Command not found
Makefile.freebsd:21: recipe for target 'libibooter.o' failed
make: *** [libibooter.o] Error 127
```
Perhaps installing `g++-8` would fix this but that package is **not** part of _build-essential_.

`make -f Makefile.linux` result is:

```
rm -rf build
mkdir -p build; cp *dat build/; cp -r docs build/
g++ -g -c libibooter.cpp -Isrc -o build/libibooter.o
g++ -g -c ziphone.cpp -Isrc -o build/ziphone.o
cd build; g++ -lusb libibooter.o ziphone.o -o ziphone; rm -rf *.o
libibooter.o: In function `ibooter::CIBootConn::CIBootConn(int, int)':
/home/dotnet/projects/playground/ZiPhone2/libibooter.cpp:25: undefined reference to `usb_init'
/home/dotnet/projects/playground/ZiPhone2/libibooter.cpp:26: undefined reference to `usb_find_busses'
/home/dotnet/projects/playground/ZiPhone2/libibooter.cpp:27: undefined reference to `usb_find_devices'
libibooter.o: In function `ibooter::CIBootConn::FindDevice(int, int) const':
/home/dotnet/projects/playground/ZiPhone2/libibooter.cpp:40: undefined reference to `usb_get_busses'
libibooter.o: In function `ibooter::CIBootConn::Connect()':
/home/dotnet/projects/playground/ZiPhone2/libibooter.cpp:60: undefined reference to `usb_open'
/home/dotnet/projects/playground/ZiPhone2/libibooter.cpp:65: undefined reference to `usb_set_configuration'
/home/dotnet/projects/playground/ZiPhone2/libibooter.cpp:71: undefined reference to `usb_claim_interface'
libibooter.o: In function `ibooter::CIBootConn::Disconnect()':
/home/dotnet/projects/playground/ZiPhone2/libibooter.cpp:91: undefined reference to `usb_release_interface'
libibooter.o: In function `ibooter::CIBootConn::WriteFile(char*, int&)':
/home/dotnet/projects/playground/ZiPhone2/libibooter.cpp:214: undefined reference to `usb_bulk_write'
libibooter.o: In function `ibooter::CIBootConn::ReadFile(char*, int&)':
/home/dotnet/projects/playground/ZiPhone2/libibooter.cpp:224: undefined reference to `usb_bulk_read'
libibooter.o: In function `ibooter::CIBootConn::ReadControl(ibooter::CIBootConn::SMessage*)':
/home/dotnet/projects/playground/ZiPhone2/libibooter.cpp:233: undefined reference to `usb_interrupt_read'
libibooter.o: In function `ibooter::CIBootConn::WriteControl(ibooter::CIBootConn::SMessage*)':
/home/dotnet/projects/playground/ZiPhone2/libibooter.cpp:241: undefined reference to `usb_interrupt_write'
libibooter.o: In function `ibooter::CIBootConn::WriteSerial(char*, int&)':
/home/dotnet/projects/playground/ZiPhone2/libibooter.cpp:332: undefined reference to `usb_bulk_write'
libibooter.o: In function `ibooter::CIBootConn::ReadSerial(char*, int&)':
/home/dotnet/projects/playground/ZiPhone2/libibooter.cpp:342: undefined reference to `usb_bulk_read'
libibooter.o: In function `ibooter::CIBootConn::CloseUsb()':
/home/dotnet/projects/playground/ZiPhone2/libibooter.cpp:351: undefined reference to `usb_close'
collect2: error: ld returned 1 exit status
mv build ziphone-3.6-cli-linux; tar -jcvf ziphone-3.6-freebsd-cli.tar.bz2 ziphone-3.6-linux-cli
tar: ziphone-3.6-linux-cli: Cannot stat: No such file or directory
tar: Exiting with failure status due to previous errors
Makefile.linux:31: recipe for target 'bzip2' failed
make: *** [bzip2] Error 2
```

_build-essential_ dependents are: 
```
Depends: libc6-dev | libc-dev, gcc (>= 4:7.2), g++ (>= 4:7.2), make, dpkg-dev (>= 1.17.11)
```

After installing _g++-8_ and changing `CC=g++8` in [Makefile.freebsd](../blob/d7dca81b2707fe962f116b23bface18de88f4351/Makefile.freebsd#L3) to `CC=g++-8`, I get the following (similar to _Makefile.linux_) result:

`make -f Makefile.freebsd`:
```
rm -rf build
mkdir -p build; cp *dat build/; cp -r docs build/
g++-8 -g -c libibooter.cpp -Isrc -o build/libibooter.o
g++-8 -g -c ziphone.cpp -Isrc -o build/ziphone.o
cd build; g++-8 -lusb libibooter.o ziphone.o -o ziphone; rm -rf *.o
libibooter.o: In function `ibooter::CIBootConn::CIBootConn(int, int)':
/home/dotnet/projects/playground/ZiPhone2/libibooter.cpp:25: undefined reference to `usb_init'
/home/dotnet/projects/playground/ZiPhone2/libibooter.cpp:26: undefined reference to `usb_find_busses'
/home/dotnet/projects/playground/ZiPhone2/libibooter.cpp:27: undefined reference to `usb_find_devices'
libibooter.o: In function `ibooter::CIBootConn::FindDevice(int, int) const':
/home/dotnet/projects/playground/ZiPhone2/libibooter.cpp:40: undefined reference to `usb_get_busses'
libibooter.o: In function `ibooter::CIBootConn::Connect()':
/home/dotnet/projects/playground/ZiPhone2/libibooter.cpp:60: undefined reference to `usb_open'
/home/dotnet/projects/playground/ZiPhone2/libibooter.cpp:65: undefined reference to `usb_set_configuration'
/home/dotnet/projects/playground/ZiPhone2/libibooter.cpp:71: undefined reference to `usb_claim_interface'
libibooter.o: In function `ibooter::CIBootConn::Disconnect()':
/home/dotnet/projects/playground/ZiPhone2/libibooter.cpp:91: undefined reference to `usb_release_interface'
libibooter.o: In function `ibooter::CIBootConn::WriteFile(char*, int&)':
/home/dotnet/projects/playground/ZiPhone2/libibooter.cpp:214: undefined reference to `usb_bulk_write'
libibooter.o: In function `ibooter::CIBootConn::ReadFile(char*, int&)':
/home/dotnet/projects/playground/ZiPhone2/libibooter.cpp:224: undefined reference to `usb_bulk_read'
libibooter.o: In function `ibooter::CIBootConn::ReadControl(ibooter::CIBootConn::SMessage*)':
/home/dotnet/projects/playground/ZiPhone2/libibooter.cpp:233: undefined reference to `usb_interrupt_read'
libibooter.o: In function `ibooter::CIBootConn::WriteControl(ibooter::CIBootConn::SMessage*)':
/home/dotnet/projects/playground/ZiPhone2/libibooter.cpp:241: undefined reference to `usb_interrupt_write'
libibooter.o: In function `ibooter::CIBootConn::WriteSerial(char*, int&)':
/home/dotnet/projects/playground/ZiPhone2/libibooter.cpp:332: undefined reference to `usb_bulk_write'
libibooter.o: In function `ibooter::CIBootConn::ReadSerial(char*, int&)':
/home/dotnet/projects/playground/ZiPhone2/libibooter.cpp:342: undefined reference to `usb_bulk_read'
libibooter.o: In function `ibooter::CIBootConn::CloseUsb()':
/home/dotnet/projects/playground/ZiPhone2/libibooter.cpp:351: undefined reference to `usb_close'
collect2: error: ld returned 1 exit status
mv build ziphone-3.6-cli-freebsd-32; tar -jcvf ziphone-3.6-cli-freebsd-32.tar.bz2 ziphone-3.6-cli-freebsd-32
ziphone-3.6-cli-freebsd-32/
ziphone-3.6-cli-freebsd-32/dfu.dat
ziphone-3.6-cli-freebsd-32/igor.dat
ziphone-3.6-cli-freebsd-32/victor.dat
ziphone-3.6-cli-freebsd-32/zibri.dat
ziphone-3.6-cli-freebsd-32/docs/
ziphone-3.6-cli-freebsd-32/docs/license.txt
ziphone-3.6-cli-freebsd-32/docs/plugin-howto.txt
ziphone-3.6-cli-freebsd-32/docs/readme.txt
ziphone-3.6-cli-freebsd-32/docs/troubleshooting.txt
```

What is missing?

--- a/Makefile.linux
+++ b/Makefile.linux
@@ -28,7 +28,7 @@ setup: clean
 	mkdir -p $(OBJDIR); cp *dat $(OBJDIR)/; cp -r docs $(OBJDIR)/
 
 bzip2:
-	mv $(OBJDIR) ziphone-3.6-cli-linux; tar -jcvf ziphone-3.6-freebsd-cli.tar.bz2 ziphone-3.6-linux-cli
+	mv $(OBJDIR) ziphone-3.6-linux-cli; tar -jcvf ziphone-3.6-linux-cli.tar.bz2 ziphone-3.6-linux-cli
 
 
 clean:

--- a/Makefile.linux
+++ b/Makefile.linux
@@ -22,7 +22,7 @@ VPATH=$(SRCDIR)
 
 
 ziphone: $(LIBOBJS) $(OBJS) 
-	cd $(OBJDIR); $(CC) $(LDFLAGS) $^ -o $@; rm -rf *.o
+	cd $(OBJDIR); $(CC) $^ $(LDFLAGS) -o $@; rm -rf *.o
 
 setup: clean
 	mkdir -p $(OBJDIR); cp *dat $(OBJDIR)/; cp -r docs $(OBJDIR)/

--- a/install-linux.txt
+++ b/install-linux.txt
@@ -6,7 +6,7 @@ apt-get install build-essential libusb-dev bzip2
 
 Then cd into the directory of the extracted source and execute the command:
 
-make -f Makefile.freebsd
+make -f Makefile.linux
 
 To build the GUI, make sure the package gnome-devel is installed, build the CLI and then execute the command:
 

